frontend/src/metabase/query_builder/containers/use-register-query-builder-metabot-context.ts

```diff
@@ -1,18 +1,50 @@
 import { useRegisterMetabotContextProvider } from "metabase/metabot";
+import * as Lib from "metabase-lib";
+import type Question from "metabase-lib/v1/Question";
 
-import { getQuestion } from "../selectors";
+import { getFirstQueryResult, getQuestion } from "../selectors";
+
+export const registerQueryBuilderMetabotContextFn = ({
+  question,
+  queryResult,
+}: {
+  question: Question | undefined;
+  queryResult: any;
+}) => {
+  if (!question) {
+    return {};
+  }
+
+  const query = question.query();
+  const { isNative } = Lib.queryDisplayInfo(query);
+  const questionCtx = question.isSaved()
+    ? { id: question.id(), type: question.type() }
+    : { type: "adhoc" as const };
+  const queryCtx = {
+    query: question.datasetQuery(),
+    sql_engine: isNative ? Lib.engine(query) : undefined,
+    is_native: isNative,
```

Comment:
it appears that `query` already has a `type` of `native`, can we try to rely on that end to end instead in a follow up PR?

Reply:
Yep makes sense. I'll do a follow-up since this also needs ai-service changes.